Model with quoted variables does not simulate in OMEdit

### Description
I have a variant of the DoublePendulum model produced by OM.jl (The Julia variant of the compiler). The model is fully scalarized, similar to how base Modelica is generated in omc. 

### Steps to Reproduce
Simulate the file in OMEdit using the -d=experimental flag. 
The model passes the check stage of the frontend, but it fails in the backend. 

### Expected Behavior
I  suppose either the model should simulate or the frontend should fail earlier. 

### Screenshots

![image](https://github.com/OpenModelica/OpenModelica/assets/8775827/7240764e-f47b-49f6-88f5-01ba6b72d230)
![image](https://github.com/OpenModelica/OpenModelica/assets/8775827/728fa80f-4d8e-4b61-bc9d-7f37d4a995b0)

## Additional info 
I also attached a model that works as expected without the quoting (except for variables that are necessary to quote). I added both the backend and the frontend as components to this issue, as I am not sure if the model should pass the frontend (although I do not think it has any illegal quotes, I have not investigated that in detail). 

### Version and OS
OpenModelica 1.22.0~dev-282-g46b9098
Ubuntu-Linux 

## Files
[Pendulums.zip](https://github.com/OpenModelica/OpenModelica/files/14741194/Pendulums.zip)

> I don't have time to test this but perhaps #12139 fixes this issue.

@phannebohm  I tested it just now, and sadly it does not seem to resolve the issue. 
I suspect that it has something to do with how quoted records are generated in the C-Code

I checked it again now one error I get is: 
```
_omcQ_27MechanicsExamples_2EDoublePendulumTest_27.c:3094:7: error: request for member ‘_’ in something not a structure or union
 3094 |   tmp0._'w' = tmp2;
```
Perhaps it has something to do with quoted records using quoted fields which is used in this model in various places

--- a/OMCompiler/Compiler/Template/CodegenCFunctions.tpl
+++ b/OMCompiler/Compiler/Template/CodegenCFunctions.tpl
@@ -2087,7 +2087,7 @@ case T_COMPLEX(complexClassType = RECORD(path = path), varLst = vars) then
     <<
     <%untagTmp%> = (MMC_FETCH(MMC_OFFSET(MMC_UNTAGPTR(<%recordVar%>), <%offset%>)));
     <%unboxBuf%>
-    <%tmpVar%>._<%compname%> = <%unboxStr%>;
+    <%tmpVar%>._<%System.unquoteIdentifier(compname)%> = <%unboxStr%>;
     >>
     ;separator="\n")
   tmpVar
@@ -6338,7 +6338,7 @@ template daeExpRecord(Exp rec, Context context, Text &preExp, Text &varDecls, Te
   match rec
   case RECORD(__) then
   let name = tempDecl(underscorePath(path), &varDecls)
-  let ass = List.zip(exps,comp) |>  (exp,compn) => '<%name%>._<%compn%> = <%daeExp(exp, context, &preExp, &varDecls, &auxFunction)%>;<%\n%>'
+  let ass = List.zip(exps,comp) |>  (exp,compn) => '<%name%>._<%System.unquoteIdentifier(compn)%> = <%daeExp(exp, context, &preExp, &varDecls, &auxFunction)%>;<%\n%>'
   let &preExp += ass
   name
 end daeExpRecord;

--- a/testsuite/simulation/modelica/records/Makefile
+++ b/testsuite/simulation/modelica/records/Makefile
@@ -11,6 +11,7 @@ RecordConstructor1.mos \
 TestComplexSum1.mos \
 TestComplexSum2.mos \
 Ticket5134.mos \
+Ticket12135.mos \
 Ticket13009.mos \
 Ticket14485.mos \
 

--- a/testsuite/simulation/modelica/records/Makefile
+++ b/testsuite/simulation/modelica/records/Makefile
@@ -12,6 +12,7 @@ TestComplexSum1.mos \
 TestComplexSum2.mos \
 Ticket5134.mos \
 Ticket12135.mos \
+Ticket12135_inline.mos \
 Ticket13009.mos \
 Ticket14485.mos \
 

--- a/testsuite/simulation/modelica/records/Ticket12135.mos
+++ b/testsuite/simulation/modelica/records/Ticket12135.mos
@@ -1,0 +1,40 @@
+// name:     Ticket12135.mos
+// keywords: record quoted identifier field codegen
+// status:   correct
+// teardown_command: rm -rf _omcQ_27P_27.* output.log
+
+// Quoted record field names must be unquoted when emitted as C struct
+// member access in record literal expansion and in record constructor
+// bodies. Previously CodegenCFunctions.tpl printed the field name
+// verbatim ('w'), producing tmp._'w' = ... which is invalid C. See
+// issue #12135.
+
+loadString("
+package 'P'
+  record R
+    Real 'w';
+    Real 'T';
+  end R;
+
+  model 'M'
+    R r;
+  equation
+    r = R('w' = time, 'T' = 2*time);
+  end 'M';
+end 'P';
+"); getErrorString();
+
+simulate('P'.'M', stopTime=0.1); getErrorString();
+
+// Result:
+// true
+// ""
+// record SimulationResult
+//     resultFile = "_omcQ_27P_27._omcQ_27M_27_res.mat",
+//     simulationOptions = "startTime = 0.0, stopTime = 0.1, numberOfIntervals = 500, tolerance = 1e-6, method = 'dassl', fileNamePrefix = '_omcQ_27P_27._omcQ_27M_27', options = '', outputFormat = 'mat', variableFilter = '.*', cflags = '', simflags = ''",
+//     messages = "LOG_SUCCESS       | info    | The initialization finished successfully without homotopy method.
+// LOG_SUCCESS       | info    | The simulation finished successfully.
+// "
+// end SimulationResult;
+// ""
+// endResult

--- a/testsuite/simulation/modelica/records/Ticket12135_inline.mos
+++ b/testsuite/simulation/modelica/records/Ticket12135_inline.mos
@@ -1,0 +1,49 @@
+// name:     Ticket12135_inline.mos
+// keywords: record quoted identifier field codegen
+// status:   correct
+// teardown_command: rm -rf _omcQ_27P_27.* output.log
+
+// Companion regression test for #12135 that specifically exercises
+// CodegenCFunctions.tpl's daeExpRecord: a record literal with quoted
+// field names is inlined as a sub-expression (here, the argument to a
+// function call) rather than appearing on the right-hand side of an
+// equation. The inline path emits the struct member assignments
+// directly instead of going through the record constructor function.
+// Without the fix this produced tmp0._'w' = ... which is invalid C.
+
+loadString("
+package 'P'
+  record R
+    Real 'w';
+    Real 'T';
+  end R;
+
+  function f
+    input R rr;
+    output Real y;
+  algorithm
+    y := rr.'w' + rr.'T';
+  end f;
+
+  model 'M'
+    Real y;
+  equation
+    y = f(R('w' = time, 'T' = 2*time));
+  end 'M';
+end 'P';
+"); getErrorString();
+
+simulate('P'.'M', stopTime=0.1); getErrorString();
+
+// Result:
+// true
+// ""
+// record SimulationResult
+//     resultFile = "_omcQ_27P_27._omcQ_27M_27_res.mat",
+//     simulationOptions = "startTime = 0.0, stopTime = 0.1, numberOfIntervals = 500, tolerance = 1e-6, method = 'dassl', fileNamePrefix = '_omcQ_27P_27._omcQ_27M_27', options = '', outputFormat = 'mat', variableFilter = '.*', cflags = '', simflags = ''",
+//     messages = "LOG_SUCCESS       | info    | The initialization finished successfully without homotopy method.
+// LOG_SUCCESS       | info    | The simulation finished successfully.
+// "
+// end SimulationResult;
+// ""
+// endResult

